src/libs/OptionsListUtils.js

```diff
@@ -83,6 +84,18 @@ Onyx.connect({
         const sortedReportActions = ReportActionUtils.getSortedReportActions(_.toArray(actions), true);
         allSortedReportActions[reportID] = sortedReportActions;
         lastReportActions[reportID] = _.first(sortedReportActions);
+
+        // The report is only visible if it is the last action not deleted that
+        // does not match a closed or created state.
+        const reportActionsForDisplay = _.filter(
+            sortedReportActions,
+            (reportAction, actionKey) =>
+                ReportActionUtils.shouldReportActionBeVisible(reportAction, actionKey) &&
+                !ReportActionUtils.isWhisperAction(reportAction) &&
+                reportAction.actionName !== CONST.REPORT.ACTIONS.TYPE.CREATED &&
+                reportAction.pendingAction !== CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE,
+        );
+        visibleReportActionItems[reportID] = reportActionsForDisplay[reportActionsForDisplay.length - 1];
```

Comment:
This code is also duplicated with SidebarUtils.
Let's try to avoid duplication as much as possible

Reply:
yes, but I think that the whole code for generating could be extracted to a separate function - we would not need a separate function for getting lastActorDisplayName then. I even wanted to do this in this PR, but it's a lot of changes, so I wanted to fix the issue this way, then take care of this refactor in a follow-up. Let me know what you think about that cc @mountiny

Reply:
In this case I agree to handle this in a follow up PR to get the fix in and limit chances of regressions. Will you handle the follow up @koko57 ?

Reply:
yeah, of course! 😊